driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncMongoClient.java

```diff
@@ -34,6 +34,35 @@
 import static java.util.Objects.requireNonNull;
 
 public class SyncMongoClient implements MongoClient {
+
+    // Unfortunately this is the only way to wait for a query to be initiated, since Reactive Streams is asynchronous
+    // and we have no way of knowing. Tests which require cursor initiation to complete before execution of the next operation
+    // can set this to a positive value.  A value of 256 ms has been shown to work well.
+    // Defaults to 0
+    private static long sleepAfterCursorOpenMS;
```

Comment:
[optional]

We may use `AtomicLong.compareAndSet` to check that setters are called only if the values are 0, i.e., either after previously disabling the behavior or if it has never been enabled).

Reply:
I decided to go with simpler (to my eyes) conditional checks in the enable methods.

Reply:
The checks will definitely work, but they have less power in detecting incorrect API usage when compared to what CAS would have given us: the current checks may fail to detect situations when an "enable" operation is called concurrently with itself (the only reasonable use of the API is to call these operations sequentially).

Reply:
If we were worried about concurrency in our test runners, I would agree with you.  But I think we have far bigger problems than this if we ever decided to run multiple tests concurrently.

Reply:
Yes, that's absolutely true. I just prefer to think about APIs in isolation: the more it relies on a specific usage pattern, the more fragile the API is, provided that other usage patterns are legit.